applications/IgaApplication/custom_modelers/nurbs_geometry_modeler_sbm.cpp

```diff
@@ -134,6 +134,139 @@ void NurbsGeometryModelerSbm::CreateAndAddRegularGrid2D(
     CreateBrepsSbmUtilities.CreateSurrogateBoundary(mpSurface, surrogate_sub_model_part_inner, surrogate_sub_model_part_outer, A_uvw, B_uvw, r_iga_model_part);
 }
 
+// 3D 
+void NurbsGeometryModelerSbm::CreateAndAddRegularGrid3D( 
+    ModelPart& rModelPart,
+    const Point& A_xyz,
+    const Point& B_xyz,
+    const Point& A_uvw,
+    const Point& B_uvw,
+    const SizeType OrderU,
+    const SizeType OrderV,
+    const SizeType OrderW,
+    const SizeType NumKnotSpansU,
+    const SizeType NumKnotSpansV,
+    const SizeType NumKnotSpansW,
+    const bool AddVolumeToModelPart)
+{   
+
+    // Call the CreateAndAddRegularGrid3D method of the base class NurbsGeometryModeler
+    NurbsGeometryModeler::CreateAndAddRegularGrid3D(rModelPart, A_xyz, B_xyz,
+        A_uvw, B_uvw, OrderU, OrderV, OrderW, NumKnotSpansU, NumKnotSpansV, NumKnotSpansW, false);
+             
+    // Create the Domain/Iga Model Part
+    const std::string iga_model_part_name = mParameters["model_part_name"].GetString();
+    ModelPart& iga_model_part = mpModel->HasModelPart(iga_model_part_name)
+                                ? mpModel->GetModelPart(iga_model_part_name)
+                                : mpModel->CreateModelPart(iga_model_part_name);
+
+    // Create the True Model Part -> contains all the true boundary features
+    std::string skin_model_part_inner_initial_name = "SkinModelPartInnerInitial";
+    std::string skin_model_part_outer_initial_name = "SkinModelPartOuterInitial";
+    std::string skin_model_part_name;
+    if (mParameters.Has("skin_model_part_inner_initial_name")) {
+        skin_model_part_inner_initial_name = mParameters["skin_model_part_inner_initial_name"].GetString();
+
+        KRATOS_ERROR_IF_NOT(mpModel->HasModelPart(skin_model_part_inner_initial_name)) 
+                     << "The skin_model_part '" << skin_model_part_inner_initial_name << "' was not created in the model.\n" 
+                     << "Check the reading of the mdpa file in the import mdpa modeler."<< std::endl;
+    }
+    if (mParameters.Has("skin_model_part_outer_initial_name")) {
+        skin_model_part_outer_initial_name = mParameters["skin_model_part_outer_initial_name"].GetString();
+
+        KRATOS_ERROR_IF_NOT(mpModel->HasModelPart(skin_model_part_outer_initial_name)) 
+                     << "The skin_model_part '" << skin_model_part_outer_initial_name << "' was not created in the model.\n" 
+                     << "Check the reading of the mdpa file in the import mdpa modeler."<< std::endl;
+    }
```

Comment:
This needs to be updated accordingly ([CreateAndAddRegularGrid2d](https://github.com/KratosMultiphysics/Kratos/blob/b1c167655bfd82056c0ac76f6c96e48448631412/applications/IgaApplication/custom_modelers/nurbs_geometry_modeler_sbm.cpp#L52)).